pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -700,6 +706,48 @@ void removeSubscription(String subscriptionName) {
         return delete(false, true, false);
     }
 
+    @Override
+    public CompletableFuture<TransactionBuffer> getTxnBuffer(boolean createIfAbsent) {
+        if (buffer.isPresent()) {
+            return CompletableFuture.completedFuture(buffer.get());
+        }
+        if (createIfAbsent) {
+            return createTransactionBuffer(createIfAbsent);
+        }
+        return FutureUtil.failedFuture(new TransactionBufferNotExistOnTopicException("Not found transaction buffer"));
+    }
+
+    private CompletableFuture<TransactionBuffer> createTransactionBuffer(boolean createIfAbsent) {
+        TopicName topicName = TopicName.get(topic);
+        String txnTopic = topic + "_txnlog";
```

Comment:
I thought we are going to introduce a new `transaction` domain?

Reply:
Yeah. I will add it.